PhysicsTools/PatAlgos/python/slimming/applySubstructure_cff.py

```diff
@@ -13,13 +13,21 @@ def applySubstructure( process, postfix="" ) :
 
 
     # Configure the RECO jets
-    from RecoJets.JetProducers.ak4PFJets_cfi import ak4PFJetsPuppi
     from RecoJets.JetProducers.ak8PFJets_cfi import ak8PFJetsPuppi, ak8PFJetsPuppiSoftDrop, ak8PFJetsPuppiConstituents
+    setattr(process,'ak8PFJetsPuppi'+postfix,ak8PFJetsPuppi.clone())
+    setattr(process,'ak8PFJetsPuppiConstituents'+postfix, ak8PFJetsPuppiConstituents.clone(cut = cms.string('pt > 170.0 && abs(rapidity()) < 2.4') ))
+    setattr(process,'ak8PFJetsPuppiSoftDrop'+postfix, ak8PFJetsPuppiSoftDrop.clone( src = 'ak8PFJetsPuppiConstituents'+postfix+':constituents' ))
+    from Configuration.Eras.Modifier_run2_miniAOD_80XLegacy_cff import run2_miniAOD_80XLegacy
+    from Configuration.Eras.Modifier_run2_miniAOD_94XFall17_cff import run2_miniAOD_94XFall17
+    from Configuration.Eras.Modifier_pA_2016_cff import pA_2016
+    from Configuration.Eras.Modifier_pp_on_AA_2018_cff import pp_on_AA_2018
+    _rerun_puppijets_task = task.copy()
+    _rerun_puppijets_task.add(getattr(process,'ak8PFJetsPuppi'+postfix),
+                              getattr(process,'ak8PFJetsPuppiConstituents'+postfix),
+                              getattr(process,'ak8PFJetsPuppiSoftDrop'+postfix))
```

Comment:
Shouldn't all this ```_rerun_puppijets_task``` be done only for the empty postfix?

Also, related to https://github.com/cms-sw/cmssw/pull/29254#issuecomment-620316191, down below on L157 there is an insertion of ```ak8PFJetsPuppiSoftDropMass```, which is available in the new puppi-in-reco setup.
I think that it should be added to this ```_rerun_puppijets_task``` and removed from L157.

Reply:
> Shouldn't all this `_rerun_puppijets_task` be done only for the empty postfix?

I mean the era-driven part of it.
While for the non-empty post-fix, the modules better be added by default